server/handlers/validation_test.go

```diff
@@ -359,50 +359,161 @@ func TestValidateRootGetCurrentRootBroken(t *testing.T) {
+		if sig.KeyID == newRootID || sig.KeyID == newRootID2 {
+			sigs = append(sigs, sig)
+		}
+	}
+	require.Len(t, sigs, 2)
+	repo.Root.Signatures = sigs
+	r.Signatures = sigs
 
 	sn, err = repo.SignSnapshot(data.DefaultExpires(data.CanonicalSnapshotRole))
 	require.NoError(t, err)
-	root, targets, snapshot, timestamp, err = getUpdates(r, tg, sn, ts)
+
+	root, _, snapshot, _, err = getUpdates(r, tg, sn, ts)
 	require.NoError(t, err)
+	root.Version = repo.Root.Signed.Version
+	snapshot.Version = repo.Snapshot.Signed.Version
+	_, err = validateUpdate(serverCrypto, "testGUN", []storage.MetaUpdate{root, snapshot}, store)
+	require.NoError(t, err)
+}
 
+// A valid root rotation requires the immediately previous root ROLE be satisfied,
+// not just that there is a single root signature.  So if there were 2 keys, either
+// of which can sign the root rotation, then either one of those keys can be used
+// to sign the root rotation - not necessarily the one that signed the previous root.
+func TestValidateRootRotationMultipleKeysThreshold1(t *testing.T) {
+	repo, crypto, err := testutils.EmptyRepo("docker.com/notary")
+	require.NoError(t, err)
+	serverCrypto := copyKeys(t, crypto, data.CanonicalTimestampRole)
+	store := storage.NewMemStorage()
+
+	// add a new root key to the root so that either can sign have to sign
+	newRootKey, err := crypto.Create("root", "", data.ED25519Key)
+	require.NoError(t, err)
+	newRootID := newRootKey.ID()
+	repo.Root.Signed.Keys[newRootID] = newRootKey
+	repo.Root.Signed.Roles[data.CanonicalRootRole].KeyIDs = append(
+		repo.Root.Signed.Roles[data.CanonicalRootRole].KeyIDs, newRootID)
+
+	r, tg, sn, ts, err := testutils.Sign(repo)
+	require.NoError(t, err)
+	require.Len(t, r.Signatures, 2)
+	// make sure the old root was just signed with the first key
+	for _, sig := range r.Signatures {
+		if sig.KeyID != newRootID {
+			r.Signatures = []data.Signature{sig}
+			break
+		}
+	}
+	require.Len(t, r.Signatures, 1)
+	root, targets, snapshot, timestamp, err := getUpdates(r, tg, sn, ts)
+	require.NoError(t, err)
+
+	// set the original root in the store
 	updates := []storage.MetaUpdate{root, targets, snapshot, timestamp}
+	require.NoError(t, store.UpdateMany("testGUN", updates))
 
-	serverCrypto := copyKeys(t, crypto, data.CanonicalTimestampRole)
-	_, err = validateUpdate(serverCrypto, "testGUN", updates, store)
+	// replace the keys with just 1 key
+	newRootKey2, err := crypto.Create("root", "", data.ED25519Key)
+	require.NoError(t, err)
+	newRootID2 := newRootKey2.ID()
+	require.NoError(t, repo.ReplaceBaseKeys(data.CanonicalRootRole, newRootKey2))
+
+	r, err = repo.SignRoot(data.DefaultExpires(data.CanonicalRootRole))
+	require.NoError(t, err)
+	require.Len(t, r.Signatures, 3)
+	// delete all signatures except the one which didn't sign the previous role
+	// and the new key
+	sigs := make([]data.Signature, 0, 2)
+	for _, sig := range repo.Root.Signatures {
+		if sig.KeyID == newRootID || sig.KeyID == newRootID2 {
```

Comment:
this is an awesome test!  Just a small nit: it might be worth leaving a comment on what `newRootKey` and `newRootKey2` map to, just for future readability